src/main/java/org/sagebionetworks/bridge/spring/controllers/StudyController.java

```diff
@@ -174,19 +175,28 @@ public Study finishStudyLogo(@PathVariable String id, @PathVariable("createdOn")
     }
 
     @GetMapping(path = "/v1/apps/{appId}/studies/{studyId}", produces = { APPLICATION_JSON_UTF8_VALUE })
-    public String getStudyForApp(@PathVariable String appId, @PathVariable String studyId)
+    public String getStudyForApp(@PathVariable String appId, @PathVariable String studyId,
+            @RequestParam(required = false) String full)
             throws JsonProcessingException {
-        CacheKey key = CacheKey.publicStudy(appId, studyId);
-        String json = cacheProvider.getObject(key, String.class);
-        if (json == null) {
+        boolean fullBool = Boolean.parseBoolean(full);
+        if (fullBool) {
+            getAuthenticatedSession(WORKER);
             appService.getApp(appId);
             Study study = service.getStudy(appId, studyId, true);
-            json = Study.STUDY_SUMMARY_WRITER.writeValueAsString(study);
-            cacheProvider.setObject(key, json, ONE_DAY_IN_SECONDS);
+            return BridgeObjectMapper.get().writeValueAsString(study);
+        } else {
+            CacheKey key = CacheKey.publicStudy(appId, studyId);
+            String json = cacheProvider.getObject(key, String.class);
+            if (json == null) {
+                appService.getApp(appId);
+                Study study = service.getStudy(appId, studyId, true);
+                json = Study.STUDY_SUMMARY_WRITER.writeValueAsString(study);
+                cacheProvider.setObject(key, json, ONE_DAY_IN_SECONDS);
+            }
+            return json;
```

Comment:
Does anything use this other than a worker? I wonder if we created this then never used it as a public endpoint like this...